plugin-agent-admin-v2/src/components/AgentAdminView/AgentAdminView.jsx

```diff
@@ -12,86 +12,75 @@ import {
-    this.setState({ selectedWorker: worker });
+  const openEditWorkerAttr = (worker) => {
+    setSelectedWorker(worker);
     Actions.invokeAction('SetComponentState', {
       name: 'WorkerAttributes',
       state: { isOpen: true }
     });
 
   }
 
-  resetWorker = () => {
-    this.setState(INITIAL_STATE);
-
+  const resetWorker = () => {
+    setSelectedWorker('');
   }
 
-
-  updateTeamFilter = (e) => {
+  const updateTeamFilter = (e) => {
     const teamFilter = e.target.value.replace(/\s/g, "");
-    this.setState({ filters: { ...this.state.filters, team: teamFilter } });
+    setTeamFilterValue(teamFilter);
   };
 
-  filterTeam = (worker) => {
-    return (!this.state.filters.team ||
-      !worker.attributes.team_name ||
-      worker.attributes.team_name.includes(this.state.filters.team));
+  const filterByTeam = (worker) => {
+    return (!teamFilterValue || !worker.attributes.team_name ||
+      worker.attributes.team_name.includes(teamFilterValue));
   }
 
-  updateSkillsFilter = (e) => {
+  const updateSkillsFilter = (e) => {
     const skillsFilter = e.target.value.replace(/\s/g, "");
-    this.setState({ filters: { ...this.state.filters, skills: skillsFilter } });
+    setSkillsFilterValue(skillsFilter);
   };
 
-  filterSkills = (worker) => {
-    return (!this.state.filters.skills ||
-      !worker.attributes.skillsString ||
-      worker.attributes.skillsString.includes(this.state.filters.skills));
+  const filterBySkills = (worker) => {
+    return (!skillsFilterValue || !worker.attributes.skillsString ||
+      worker.attributes.skillsString.includes(skillsFilterValue));
   }
 
-  updateNameSort = (e) => {
-    const newSortOrder = this.state.sort.name === "asc" ? "desc" : "asc";
-    this.setState({
-      sort: { ...this.state.sort, name: newSortOrder },
-    });
+  const updateNameSort = (e) => {
+    const newSortOrder = sort.name === "asc" ? "desc" : "asc";
+    setSort({ name: newSortOrder });
   };
 
-  sortName = (v1, v2) => {
+  const sortByName = (v1, v2) => {
     const result =
-      this.state.sort.name === "asc"
+      sort.name === "asc"
         ? v1.attributes.full_name < v2.attributes.full_name ? 1 : -1
         : v1.attributes.full_name > v2.attributes.full_name ? 1 : -1;
     return result;
   };
 
-  render() {
-    const nameSortValue = this.state.sort.name;
-    const { teamFilterValue, skillsFilterValue } = this.state.filters;
+  const nameSortValue = sort.name;
 
-    const sortedWorkers = this.props.workers
-      .filter(this.filterTeam)
-      .filter(this.filterSkills)
-      .sort(this.sortName);
+  let sortedWorkers = [];
+  if (workers && workers.length > 1) {
+    sortedWorkers = workers
+      .filter(filterByTeam)
+      .filter(filterBySkills)
+      .sort(sortByName);
+  }
 
-    console.log("workers", sortedWorkers);
-    return (
-      <Theme.Provider theme="flex">
-        <Box overflow='auto' maxHeight= '100%'>
+  return (
+    <Theme.Provider theme="flex">
```

Comment:
I would use the PasteThemeProvider in the init file rather than individually wrapping each component.